### DISCUSSION TITLE
Docstring alias for modules that mix bindings and pure Python

### DISCUSSION THREAD

**Discussion by jpreiss** [pybind/pybind11#4840] [Ideas]
My project mixes pybind11 output with pure Python code. I follow the pattern in https://github.com/pybind/pybind11/issues/1004#issuecomment-322941844 and https://github.com/pybind/pybind11/issues/1004#issuecomment-1686333057: give an "ugly" name for the pybind11 module, then in the top-level module's `__init__.py` have

```
from _ugly_name import Class

__all__ = ["Class"]
```

However, the docstrings generated look like:

```
method_name(self: _ugly_name.Class) -> int
```

It would be great if I could choose a different module name.

Since pybind11 has no awareness of how the bindings fit into the larger package, we cannot hope to do this automatically. What if `pybind11::options` had some method like `set_module_doc_alias(char const *)` that would allow the user to choose how the module's name appears in generated docstrings?

**Comment by tvatter** [comment]
@jpreiss Did you find a solution to this issue?

**Reply by jpreiss** [reply]
unfortunately not :(

**Reply by tvatter** [reply]
The trick suggested [here](https://github.com/pybind/pybind11/discussions/3748) worked for me.
